cli_tools/common/utils/logging/logger.go

```diff
@@ -15,50 +15,200 @@
 package logging
 
 import (
-	"fmt"
 	"log"
+	"os"
+	"strings"
+	"sync"
 	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/proto/go/pb"
+	"github.com/golang/protobuf/proto"
 )
 
-// LogEntry encapsulates a single log entry.
-type LogEntry struct {
-	LocalTimestamp time.Time `json:"localTimestamp"`
-	Message        string    `json:"message"`
+// LogWriter is a logger for interactive tools. It supports
+// string messages and structured metrics.
```

Comment:
This PR makes a design choice: Method names indicate the destination/consumer of a message. To show a message to a user, call `WriteUser`. Other systems use generic names such as `INFO`; this PR consciously avoids that, since `INFO` is vague and requires extra work for future developers to understand where a message will be written.

Given that, `WriteMetric` signals that the destination is the metrics server.

Regarding structured messages -- the OVF consolidation project doesn't require that, so I didn't design that into the logger. From the top of my head: if we wanted to support other types in the future, we could change the parameter type to `interface {}` in the future.

Does that sound okay?

Reply:
The consumer of messages is up to clients calling our backend. gcloud and Pantheon do show user messages to users, but CLI tools could potentially be run via API or integrated into other clients where there might be no users to see these messages. Ideally, CLI tools should not be aware of gcloud and pantheon, if possible.

Reply:
> CLI tools could potentially be run via API or integrated into other clients where there might be no users to see these messages. Ideally, CLI tools should not be aware of gcloud and pantheon, if possible.

I like that you're thinking of future extensions. Would you be alright trying this interface for our current requirements? In the future, if this code is deployed such that there isn't an end user, then we can re-evaluate whether the interface makes sense?